core/src/actions/schedules.ts

```diff
@@ -211,21 +211,25 @@ export class ScheduleEdit extends AuthenticatedAction {
   async runWithinTransaction({ params }: { params: ParamsFrom<ScheduleEdit> }) {
     const schedule = await Schedule.findById(params.id);
 
-    const recurringFrequency =
-      params.recurring && params.recurringFrequency
-        ? params.recurringFrequency
-        : 0;
-
-    // these timing options are validated separately, and should be set first
-    await schedule.update({
-      recurring: !!params.recurring,
-      recurringFrequency,
-    });
+    if (typeof params.recurring === "boolean") {
+      const recurringFrequency =
+        params.recurring && params.recurringFrequency
+          ? params.recurringFrequency
+          : 0;
+
+      // these timing options are validated separately, and should be set first
+      await schedule.update({
+        recurring: params.recurring,
+        recurringFrequency,
+      });
```

Comment:
From what I'm reading, we would only be updating `recurringFrequency` if `recurring` is passed at the same time. Could there be an issue here if you don't touch the `recurring` field on the form but do want to change the frequency?

Reply:
Yep, this is how it was previously set before #2828, and wanted to see if this is what was intended. However, one change I did in that PR is that you can only set `recurringFrequency` > 0 if `recurring` is `true`.